### PR TITLE
Add new job to testing PR check that tests with latest SNAPSHOT version of Drools

### PR DESCRIPTION
**Pull request by baldimir**
Hi, sometimes, there may be new FEEL functionality added in the Drools engine, e.g. as part of implementing new DMN versions. This new functionality may not yet be available in a stable Drools version and it may only be in the SNAPSHOT versions released. With this, writing new documentation may be not possible, because there is no way, how to test new FEEL expressions added, that depend on functionality in the SNAPSHOTS. 

This PR adds another job to the test workflow, which runs all the expressions against the SNAPSHOT version of Drools. There are two PR checks scenarios then (apart of scenarios due to failing expressions because they are written incorrectly): 
- New chapters/FEEL expressions added, that rely on SNAPSHOTs: 
  - This shows up as a failure on the Drools stable test job and shows the SNAPSHOT test job green. 
- Changes done using the existing Drools DMN functionality:
  - Both jobs are green. 
 
@tarilabs what do you think about this please? Would it be fine, to add another job to the PR checks please?

### CODE REVIEW

**Review by gitgabrio — APPROVED** [review]
Thanks. LGTM

**Comment by tarilabs** [comment]
> @tarilabs what do you think about this please? Would it be fine, to add another job to the PR checks please?

Ciao @baldimir thanks for this PR. The additional CI check is fine, it could also support as an additional check of non-regression for the forward development.

I'm doubtful however of one implication: now you end up with example expressions in the Doc/Manual which do not (yet) work with a Released version, which might confuse the reader.

In this case, I believe the correct action would have been to make this manual DMN-version dependent (not Drools-version dependent).

Hope this helps? lmk

**Comment by gitgabrio** [comment]
Hi @tarilabs
Thanks for suggestion! We touched the topic of versioned-handbook some times ago with  @baldimir @yesamer and others, but did not have time, yet, to actually put in place.

**Review by yesamer — APPROVED** [review]

**Comment by baldimir** [comment]
Yes, the next step is to make this versioned based on the DMN versions.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -23,3 +23,20 @@ jobs:
...
+        with:
+          path: /root/.jbang
+          key: $-jbang-$
+          restore-keys: |
+            $-jbang-
+      - name: jbang
+        uses: jbangdev/jbang-action@v0.111.0
+        with:
+          script: test.java
+          jbangargs: -Ddrools.version=999-SNAPSHOT
```
Comment (tarilabs) [inline .github/workflows/test.yml:42]: for the next iteration, you can consider making this a GHA-matrix parameter and define a parametrized GHA-job

**Review by tarilabs — APPROVED** [review]

**Pull request merged**